app/api/dao/user.py

```diff
@@ -153,7 +153,11 @@ def list_users(user_id, is_verified=None):
                     list_of_users += [user.json()]
         else:
             list_of_users = [user.json() for user in users_list]
-
+        # user is available for relation if: not in relation * (need_mentoring + available_to_mentor)
+        for user in list_of_users:
+            user['is_available'] = False
+            if user['membership_status'] == None and (user['need_mentoring'] or user['available_to_mentor']):
```

Comment:
Is it one to one relation? One mentor can't have more than one mentees?

Reply:
membership_status field was not giving proper value when member was in mentorship relation. I replaced this with a call to ```MentorshipRelationDAO.list_current_mentorship_relation```. Boolean logic remains the same

```py
            current_relation_response = MentorshipRelationDAO.list_current_mentorship_relation(user['id'])
            # find if not in relation
            if not isinstance(current_relation_response,MentorshipRelationModel):
                if (user['need_mentoring'] or user['available_to_mentor']):
                    user['is_available'] = True
```